### PR TITLE
Run GitHub Actions locally

### PR DESCRIPTION
**Pull request by lslezak**
## New GitHub Actions Tasks

See the details in the updated `README.md` file.

- `actions:list` - print the defined GitHub Action jobs
- `actions:details` - print the details of the defined GitHub Action jobs.
- `actions:run[job]` - run the specified  GitHub Action job locally, if not specified then it runs all jobs
<br>

- `run:container[client]`  - this is similar to the `run[client]` task, but it runs the client in a Docker container. Because the Docker container environment is quite limited (no systemd, no boot loader, no HW access) some modules will not work properly. But it works for some modules/clients (like `repositories`) so let's try it.

## Testing

You can run
```console
rake actions:run
```
or if you use system with older Docker which does not support new glibc-2.33 in openSUSE TW/Factory then run
```console
rake actions:run DOCKER_IMAGE=registry.opensuse.org/yast/head/containers_15.3/yast-ruby:latest
```
to force using a different (Leap 15.3) Docker image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,31 @@ for more details.
 
 ## run[client]
 Runs client with paths leading to git. Useful to testing module without
-installation.
+installation. If the client is not specified it starts the client
+with the shortest name.
+
+## run:container[client]
```
Comment (jreidinger) [inline README.md:45]: :+1: good idea

File: lib/tasks/actions.rake [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  desc "Display GitHub Action job details"
+  task :details do
+    GithubActions::Tasks::Details.new.run
+  end
+
+  desc "Run the specified GitHub Action job locally"
+  task :run, [:job] do |_task, args|
+    name = args[:job]
+    abort "ERROR: Missing job name argument" if name.nil? || name.empty?
```
Comment (jreidinger) [inline lib/tasks/actions.rake:38]: what about reducing number of tasks and when name is not specified run all of them?

Comment (lslezak) [reply lib/tasks/actions.rake:38]: OK, that sounds good...

**Comment by jreidinger** [comment]
Overall impression is that it looks really good and useful not just for yast. I expect it can be used also in libyui and other projects using github actions. I think it makes sense to have it as separate rubygem and just add it as depedency of yast-rake and libyui-rake and maybe make a bit of publicity around, as other projects can find it useful.

**Comment by lslezak** [comment]
At the beginning I was thinking about to make it a separate gem. But currently it is highly tight to YaST, it supports only the GitHub Actions features which *we* use. And that's a very small subset.

On the other hand there is the [act](https://github.com/nektos/act) project which supports much more GitHub Actions features (all?). So it does not make much sense to create a parallel project.

Let's keep it here for now and get some feedback. If it turns out to be useful and nice to have also in libyui then we can move it later.

**Comment by mvidner** [comment]
Nice! This PR can be easily tested with `rake -f ../yast-rake/Rakefile ...` and I did and it works :+1:

File: lib/tasks/github_actions/github_actions/workflow.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      end
+    end
+
+    # load the workflow from an YAML file
+    # @param file [String] path to the YAML file
+    def initialize(file)
+      @file = file
+      yml = YAML.load_file(file)
+      @name = yml["name"]
+      # "on" is autoconverted to Boolean "true"!
```
Comment (mvidner) [inline lib/tasks/github_actions/github_actions/workflow.rb:43]: This sounds boring until you realize we're talking about `on: [push, pull_request]`

:facepalm:

**Review by mvidner — APPROVED** [review]
Wow!

**Comment by jreidinger** [comment]
@lslezak if there is already act gem, why not just use it? It can save some time and we do not need to maintain that local runner ourself, not?

**Comment by mvidner** [comment]
@jreidinger "act" is not a gem, it is written in golang. But yes, that means it is easy to run: git clone, zypper in go, make, dist/local/act --help

File: lib/tasks/actions.rake [hunk 4]
```diff
@@ -35,13 +35,10 @@ namespace :actions do
...
     name = args[:job]
-    abort "ERROR: Missing job name argument" if name.nil? || name.empty?
-
-    GithubActions::Tasks::Run.new(name).run
-  end
-
-  desc "Run all supported GitHub Action jobs locally"
-  task :"run:all" do
-    GithubActions::Tasks::RunAll.new.run
+    if name.nil? || name.empty?
```
Comment (jreidinger) [inline lib/tasks/actions.rake:38]: I would also update description like `"Run the GitHub Actions jobs or specified one locally"

**Review by jreidinger — APPROVED** [review]

**Pull request merged**